### PR TITLE
[GLUTEN-1476][VL] Support GetStructField

### PR DESCRIPTION
**Pull request by rui-mo**
## What changes were proposed in this pull request?

Support GetStructField on Velox backend.
https://github.com/oap-project/velox/pull/294

https://github.com/oap-project/gluten/issues/1476

## How was this patch tested?

under test

### CODE REVIEW

**Comment by rui-mo** [comment]
Depends on https://github.com/oap-project/gluten/pull/1878.

File: backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala [hunk 1]
```diff
@@ -353,7 +353,11 @@ class VeloxDataTypeValidationSuite extends WholeStageTransformerSuite {
     // Validation: BatchScan Project Aggregate Sort Limit
     // TODO validate Expand operator support Struct type ?
     runQueryAndCompare("select int, struct.struct_1 from type1 " +
-      "sort by struct.struct_1 limit 1") { _ => }
+      "sort by struct.struct_1 limit 1") { df => {
```
Comment (JkSelf) [inline backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala:356]: Do we need to add the nested struct type test？

Comment (rui-mo) [reply backends-velox/src/test/scala/io/glutenproject/execution/VeloxDataTypeValidationSuite.scala:356]: There are several tests from Spark on that in [GlutenDataFrameSetOperationsSuite](https://github.com/oap-project/gluten/blob/main/gluten-ut/spark32/src/test/scala/org/apache/spark/sql/GlutenDataFrameSetOperationsSuite.scala) and were fixed.

**Review by JkSelf — APPROVED** [review]
LGTM.

**Comment by rui-mo** [comment]
@JkSelf Could you help review agian? Just reverted the test branch.

**Review by JkSelf — APPROVED** [review]

**Pull request merged**